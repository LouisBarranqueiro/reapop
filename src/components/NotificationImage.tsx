import React from 'react'
import {Notification} from '..'
import {classnames} from '../constants'
import {useTheme} from '../hooks/useTheme'

type Props = {
    notification: Notification
}

const NotificationImage = (props: Props) => {
    const {notification} = props
    const theme = useTheme()
    const containerStyles = theme ? theme.notificationImageContainer(notification) : {}
    const imageStyles = theme ? theme.notificationImage(notification) : {}

    return (
        <div className={classnames.notificationImageContainer} style={containerStyles}>
            <span
                className={classnames.notificationImage}
                style={{
                    ...imageStyles,
                    ...{backgroundImage: `url(${notification.image as string})`},
                }}
            />
        </div>
    )
}

export default NotificationImage
