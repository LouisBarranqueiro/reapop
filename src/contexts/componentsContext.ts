import {createContext} from 'react'
import NotificationIcon from '../components/NotificationIcon'
import Notification from '../components/Notification'
import SlideTransition from '../components/SlideTransition'
import GrowTransition from '../components/GrowTransition'
import FadeTransition from '../components/FadeTransition'

export type ComponentContextType = {
    Transition?: typeof SlideTransition | typeof GrowTransition | typeof FadeTransition
    NotificationIcon?: typeof NotificationIcon
    Notification?: typeof Notification
}

export const ComponentsContext = createContext<ComponentContextType>({})
