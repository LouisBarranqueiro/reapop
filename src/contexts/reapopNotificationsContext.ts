import {createContext} from 'react'
import {NewNotification, Notification} from '../reducers/notifications/types'

type NotificationsContext = {
    notifications: Notification[]
    notify: (notification: NewNotification) => void
    dismissNotification: (id: string) => void
    dismissNotifications: () => void
}

export const ReapopNotificationsContext = createContext<NotificationsContext>(undefined!)
