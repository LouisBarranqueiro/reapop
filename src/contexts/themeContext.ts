import {createContext} from 'react'
import {Theme} from '..'

export const ThemeContext = createContext<Theme | undefined>(undefined)
