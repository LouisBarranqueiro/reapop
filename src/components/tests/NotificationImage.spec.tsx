import React from 'react'
import {render} from '@testing-library/react'

import {atalhoTheme, POSITIONS, STATUSES} from '../..'
import NotificationImage from '../NotificationImage'
import {ThemeContext} from '../../contexts/themeContext'

describe('<NotificationImage/>', () => {
    beforeEach(() => {
        jest.clearAllMocks()
        jest.spyOn(atalhoTheme, 'notificationImageContainer').mockReturnValue({color: 'green'})
        jest.spyOn(atalhoTheme, 'notificationImage').mockReturnValue({color: 'white'})
    })

    afterAll(() => {
        jest.resetAllMocks()
    })

    it.each([
        ['without theme', undefined],
        ['with theme', atalhoTheme],
    ])('should display notification image %s', (_, theme) => {
        const notification = {
            id: '1',
            status: STATUSES.info,
            position: POSITIONS.topLeft,
            image: 'image.png',
            buttons: [],
        }
        const {container} = render(
            <ThemeContext.Provider value={theme}>
                <NotificationImage notification={notification} />
            </ThemeContext.Provider>
        )
        expect(container.innerHTML).toMatchSnapshot()
        expect((atalhoTheme.notificationImageContainer as jest.Mock).mock.calls).toMatchSnapshot()
        expect((atalhoTheme.notificationImage as jest.Mock).mock.calls).toMatchSnapshot()
    })
})
