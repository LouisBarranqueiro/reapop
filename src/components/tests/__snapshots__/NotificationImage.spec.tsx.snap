// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NotificationImage/> should display notification image with theme 1`] = `"<div class=\\"reapop__notification-image-container\\" style=\\"color: green;\\"><span class=\\"reapop__notification-image\\" style=\\"color: white; background-image: url(image.png);\\"></span></div>"`;

exports[`<NotificationImage/> should display notification image with theme 2`] = `
Array [
  Array [
    Object {
      "buttons": Array [],
      "id": "1",
      "image": "image.png",
      "position": "top-left",
      "status": "info",
    },
  ],
]
`;

exports[`<NotificationImage/> should display notification image with theme 3`] = `
Array [
  Array [
    Object {
      "buttons": Array [],
      "id": "1",
      "image": "image.png",
      "position": "top-left",
      "status": "info",
    },
  ],
]
`;

exports[`<NotificationImage/> should display notification image without theme 1`] = `"<div class=\\"reapop__notification-image-container\\"><span class=\\"reapop__notification-image\\" style=\\"background-image: url(image.png);\\"></span></div>"`;

exports[`<NotificationImage/> should display notification image without theme 2`] = `Array []`;

exports[`<NotificationImage/> should display notification image without theme 3`] = `Array []`;
